Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,469 @@
+                if (item.Status != OrderStatus.Canceled && item.Status != OrderStatus.Filled && item.Status != OrderStatus.Invalid)
+                {
+                    var cached = this.CachedOrderIDs.Where(c => c.Value.BrokerId.Contains(item.BrokerId.First()));
+                    if (cached.Count() > 0 && cached.First().Value != null)
+                    {
+                        this.CachedOrderIDs[cached.First().Key] = item;
+                    }
+                }
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Retreive holdings from exchange
+        /// </summary>
+        /// <returns></returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+
+            var list = new List<Holding>();
+
+            var response = RestClient.GetActivePositions();
+            foreach (var item in response)
+            {
+                var ticker = RestClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
+                list.Add(new Holding
+                {
+                    Symbol = Symbol.Create(item.Symbol, SecurityType.Forex, Market.Bitfinex),
+                    Quantity = decimal.Parse(item.Amount) * divisor,
+                    Type = SecurityType.Forex,
+                    CurrencySymbol = "B",
+                    ConversionRate = (decimal.Parse(ticker.Mid) / divisor),
+                    MarketPrice = (decimal.Parse(ticker.Mid) / divisor),
+                    AveragePrice = (decimal.Parse(item.Base) / divisor),
+                });
+            }
+            return list;
+        }
+
+
+        /// <summary>
+        /// Get Cash Balances from exchange
+        /// </summary>
+        /// <returns></returns>
+        //todo: handle other currencies
+        public override List<Securities.Cash> GetCashBalance()
+        {
+            var list = new List<Securities.Cash>();
+            var response = RestClient.GetBalances();
+            foreach (var item in response)
+            {
+                if (item.Type == wallet)
+                {
+                    if (item.Currency == "usd")
+                    {
+                        list.Add(new Securities.Cash(item.Currency, item.Amount, 1));
+                    }
+                    else
+                    {
+                        var ticker = RestClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
+                        list.Add(new Securities.Cash("BTC", item.Amount * divisor, decimal.Parse(ticker.Mid) / divisor));
+                    }
+                }
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Get queued tick data
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<Data.BaseData> GetNextTicks()
```

Comment:
for cleanliness, we may want to split the `IDataQueueHandler` implementation into a separate partial class file, like was done with the`FxcmBrokerage`